File: ToDo_Backend/src/main/java/com/Daniel/ToDo_Backend/service/TaskService.java
package com.Daniel.ToDo_Backend.service;

import com.Daniel.ToDo_Backend.model.Task;

import java.util.List;

public interface TaskService {

    public Task saveTask(Task task);

    public List<Task> getAllTask();

    public Task getTaskById(Long id);

    public String deleteTask(Long id);

    public Task editTask(Task task, Long id);
}

File: todo_frontend/src/service/TaskService.js
import axios from "axios";

const API_URL = "http://localhost:8080"; 

class TaskService {

    saveTask(task) {
        return axios.post(API_URL + "/addTask", task);
    }

    getAllTask () {
        return axios.get(API_URL + "/allTasks");
    }

    getTaskById(id) {
        return axios.get(API_URL + "/Task/" + id);
    }

    deleteTask(id) {
        return axios.delete(API_URL + "/Delete/" + id);
    }

    editTask(task) {
        return axios.put(API_URL + "/edit/" + task.id, task);
    }

}

export default new TaskService;